metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -2735,20 +2723,38 @@ class MetalsLanguageServer(
       // (require ./mill or ./.mill-version)
       buildTools.isMill
 
+  /**
+   * In case of Scala-Cli we can either import a single file or an entire directory
+   * However, we have to unsure that there is no clashes with other existing sourceItems
+   */
+  private def scalaCliDirOrFile(path: AbsolutePath): AbsolutePath = {
+    val dir = path.parent
+    val nioDir = dir.toNIO
+    val conflictsWithMainBsp =
+      buildTargets.sourceItems.filter(_.exists).exists { item =>
+        val nioItem = item.toNIO
+        nioDir.startsWith(nioItem) || nioItem.startsWith(nioDir)
```

Comment:
`nioDir.startsWith(nioItem)` why this is needed?

Does `buildTargets.sourceItems` contains directory? if yes, does that mean we can't import the entire directory for `scripts` directory? 

```
src
`-- main
    `-- scala
        |-- scripts
        |       `-- foo.sc
        `-- main.scala
```

Reply:
I think it's fine not to allow it for now. It's a very niche scenario that someone would mix source directories and script directories.

Reply:
@tanishiking 
yep in your case we won't import it.

I added this clause mainly for the following scenario:
```
workspace/
   - src/main/scala/...
   - my_script.sc   // <- if we import as `workspace` then scala-cli takes sources from /src/main/scala too
   - build.sbt
```